code/headers/st7735.hpp

```diff
@@ -0,0 +1,166 @@
+#ifndef R2D2_ST7735_HPP
+#define R2D2_ST7735_HPP
+
+#pragma once
+
+#include <hwlib.hpp>
+#include <display_adapter.hpp>
+
+namespace r2d2::display {
+
+    class st7735_c : public display_c {
+    protected:
+        // all the commands for the display
+        constexpr static uint8_t _NOP = 0x00;
```

Comment:
For the SSD1306 a enumeration is used; using that would be more consistent. I won't  disapprove the PR because of this though.